### PR TITLE
Convert rest of JIT icall registration to static info.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/exceptions-amd64.c [hunk 1]
```diff
@@ -925,16 +925,24 @@ mono_amd64_get_exception_trampolines (gboolean aot)
 	MonoTrampInfo *info;
 	GSList *tramps = NULL;
 
+	// FIXME more in the macro
+#define TRAMPOLINE(name) (info->jit_icall_info = &mono_jit_icall_info.mono_ ## name)
+
```
Comment (vargaz) [inline mono/mini/exceptions-amd64.c]: These macros are not needed.

Comment (jaykrell) [reply mono/mini/exceptions-amd64.c]: I'd like to expand them to have "every" line per trampoline.
I had that at first, but something wasn't working so I dialed back.
But ok either way.

Comment (jaykrell) [reply mono/mini/exceptions-amd64.c]: I remove all the `#define TRAMPOLINE`. There were several and they varied some. I'd like to put them back and have them do more, another time.

**Comment by jaykrell** [comment]
@monojenkins squash

**Comment by jaykrell** [comment]
One nagging sigsegv I was seeing before, I thought it'd gone away and haven't been able to repro in WSL, in Linux x64 FullAOT.
Will poke at it more and eventually use real Linux VM.

**Comment by jaykrell** [comment]
No repro with innocuous debugging code, trying again w/o. Maybe unrelated flake.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 FullAOT

**Comment by jaykrell** [comment]
@monojenkins squash

**Comment by jaykrell** [comment]
@monojenkins build Linux x64 FullAOT

**Comment by jaykrell** [comment]
3 successes in a row, I think ok & ready

**Comment by jaykrell** [comment]
@monojenkins squash

**Comment by jaykrell** [comment]
@ci-sheriffs please squash
There were some breaks but looks like unrelated flake.

**Pull request merged**